stackslib/src/chainstate/stacks/boot/pox-4.clar

```diff
@@ -658,6 +664,28 @@
 
       (ok true)))
 
+;; Verify a signature from the delegate that approves this specific stacker.
+;; The message hash is the sha256 of the consensus hash of the tuple 
+;; `{ stacker, reward-cycle }`. Note that `reward-cycle` corresponds to the
+;; _current_ reward cycle, not the reward cycle at which the delegation will start.
+;; The public key is recovered from the signature and compared to the pubkey hash
+;; of the delegator.
+(define-read-only (verify-delegator-signature (stacker principal)
+                                             (delegator principal)
+                                             (delegator-sig (buff 65)))
+  (let
+    (
+      (msg { stacker: stacker, reward-cycle: (current-pox-reward-cycle) })
+      (msg-bytes (unwrap! (to-consensus-buff? msg) ERR_DELEGATION_INVALID_SIGNATURE)) ;;TODO
+      (msg-hash (sha256 msg-bytes))
```

Comment:
We use sha512/256 for our message digests

Reply:
As in sha512(sha256(message))? No problem, I'll change that.

Reply:
I also notice that in [SIP018](https://github.com/stacksgov/sips/blob/main/sips/sip-018/sip-018-signed-structured-data.md#formal-specification) (signed structured data) only `sha256` is used, and I'd like to build this off of SIP018.

Reply:
The latest in this PR does follow SIP18, the v1 definitely didn't